kernel/kernel-api/src/main/java/io/delta/kernel/Table.java

```diff
@@ -50,14 +48,6 @@ static Table forPath(TableClient tableClient, String path)
     Snapshot getLatestSnapshot(TableClient tableClient)
         throws TableNotFoundException;
 
-    /**
-     * The fully qualified path of this {@link Table} instance.
-     *
-     * @return the table path
-     * @since 3.2.0
-     */
-    String getPath();
```

Comment:
+1 I'm also good with adding `tableClient` as an argument. Since we transform the original input path (by this I mean just qualifying it) it seems like we should provide a way for the connector to get the actual path we are using for all our ops

Reply:
If we allow creating a `Table` object for a non-existent path though it seems like these semantics are confusing

Reply:
I think having
- `getPath: String`
- `tableExists: boolean`
would solve all of our semantic issues, right?

Reply:
I can add back the `getPath(TableClient)`. This is not an issue.

Reply:
added this API back with `TableClient` as a parameter